internal/services/sentinel/sentinel_metadata_resource.go

```diff
@@ -0,0 +1,854 @@
+					},
+
+					"link": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"dependency": {
+			Type:             pluginsdk.TypeString,
+			Optional:         true,
+			ValidateFunc:     validation.StringIsJSON,
+			DiffSuppressFunc: pluginsdk.SuppressJsonDiff,
+		},
+
+		"category": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"domains": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+					"verticals": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+		},
+
+		"providers": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"first_publish_date": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validate.ISO8601DateTime,
+		},
+
+		"last_publish_date": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validate.ISO8601DateTime,
+		},
+
+		"content_schema_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Default:      "2.0",
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"custom_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"icon_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"preview_images": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"preview_images_dark": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"threat_analysis_tactics": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						"Reconnaissance",
+						"ResourceDevelopment",
+						"InitialAccess",
+						"Execution",
+						"Persistence",
+						"PrivilegeEscalation",
+						"DefenseEvasion",
+						"CredentialAccess",
+						"Discovery",
+						"LateralMovement",
+						"Collection",
+						"CommandAndControl",
+						"Exfiltration",
+						"Impact",
+						"ImpairProcessControl",
+						"InhibitResponseFunction",
```

Comment:
@ziyeqf is there a Swagger source for these?

Reply:
no, it comes from service validation rule.. just opened an issue requesting for it.https://github.com/Azure/azure-rest-api-specs/issues/22956